programs/ssl/ssl_server2.c

```diff
@@ -587,7 +588,7 @@ int main(void)
     " acceptable ciphersuite names:\n"
 
 #define ALPN_LIST_SIZE    10
-#define CURVE_LIST_SIZE   20
+#define CURVE_LIST_SIZE   25
```

Comment:
Pre-existing: the same macro is present in `ssl_client2.c`, and the changes below look identical to those made to `ssl_client2.c`, so perhaps we should take this as an indication to move the common bits to `ssl_test_lib.[ch]`?

Reply:
It can be common if the list is always the same on client and server side.

Reply:
Yes, I think the default list is always the same on both sides - it's all groups that are supported. Looking at both C files, we have 56 lines that are identical:
```c
#if defined(MBEDTLS_ECP_LIGHT)
    if (opt.curves != NULL) {
        // --snip--
   }
#endif /* MBEDTLS_ECP_LIGHT */
```
Also, this makes me notice: the guard is wrong now, should be `ECP_LIGHT || (TLS1_3 && PSA_WANT_ALG_FFDH && (EPHEMERAL || PSK_EPHEMERAL))`. Which is obviously too long, we should have a helper macro for the 2nd part (perhaps in `ssl.h` or `ssl_ciphersuites.h`), as it's probably going to be used in other places as well.

Reply:
Note: thanks for moving the macro definitions, but can you also look at the 56 lines of duplicated C code? If there's no obstacle, it would be good to de-duplicate them. If turns out to be complicated, we can leave that for a follow-up, but it's worth trying first.